src/ORM/ManyManyThroughList.php

```diff
@@ -145,14 +161,18 @@ public function add($item, $extraFields = [])
         if (is_numeric($item)) {
             $itemID = $item;
         } elseif ($item instanceof $this->dataClass) {
+            /** @var DataObject $item */
+            if (!$item->isInDB()) {
+                $item->write();
```

Comment:
Will this and `$hasManyList->add($record);` (L217) result in duplicate writes?

Reply:
No because this writes the RHS record. The manyManyList writes the join record. Separate models ;)